protocol.md

```diff
@@ -1,65 +1,129 @@
+  do not send reports.
+- **Receiver Privacy**: Users who receive reports do not reveal information to
+  anyone.
+- **Weak Reporter Privacy**: Users who send reports do not reveal information
+  to users they did not come in contact with, and reveal only the time of
+  contact to users they did come in contact with.  Note that in practice, the
+  timing alone may still be sufficient for their contact to learn their
+  identity (e.g., if their contact was only around one other person at the
+  time).
+
+**XXX** It may be possible / better to merge this with the section above.
+
+## A strawman protocol
+
+As a first attempt to formulate a protocol that satisfies these properties, we
+consider a strawman protocol.  All mobile devices running the app periodically
+generate a random CEN, store the CEN, and broadcast it using Bluetooth. At the
+same time, the app also listens for and records the CENs generated by other
+devices.  To send a report, the user (or a health authority acting on their
+behalf) uploads the CENs she generated to a server.  All users' apps
+periodically download the list of reported CENs, then compare it with the list
+of CENs they observed and recorded locally. The intersection of these two lists
+is the set of positive contacts.
+
+Intuitively, this provides server privacy, as the server only observes a list
+of random numbers, and cannot correlate them with users or locations without
+colluding with other users.  It prevents passive tracking, because all
+identifiers are randomly generated and therefore unlinkable from each other.
+It provides receiver privacy, because all users download the same list of
+reported CENs and process it locally.  And if the list of CENs is batched
+appropriately, users who send reports do not leak information beyond the time
+of contact to users who observed the CENs.
+
+However, this proposal does not provide source integrity.  Because CENs have no
+structure, nothing prevents a user from observing the CENs broadcast by another
+user and then including them in a report to the server.  Notice that this is
+still a problem even in the setting where a health authority verifies reports,
+because although they can attest to test results, they have no way to verify
+the CENs.  It also poses scalability issues, because all users must download
+the entire list of reported CENs.
+
+Both of these issues can be addressed by deriving CENs differently, as
+described below.
 
 ## Compressing CEN uploads for scalability:
 
-Although completely random CENs are appealing from a privacy perspective, they
-result in a potentially prohibitive amount of data uploaded and downloaded by
-the app. In a naive setup without sharding by location, we might be talking
-about terabytes of data ( FIXME LINK TO SPREADSHEETs with calculations). While
-location based sharding is a possibility for future versions of the protocol,
-at present it raises too many open questions. 
```

Comment:
The note about scalability is included above, and I dropped the note about location based sharding because the planning is already underway.